tests/support/lib/random_buffer.c

```diff
@@ -7,65 +7,45 @@
 #include <stdint.h>
 #include <string.h>
 
+#include <sys/param.h>
 #include <sys/types.h>
 
 #include <hse/util/xrand.h>
 
 #include <hse/test/support/random_buffer.h>
 
 void
-randomize_buffer(void *buf, size_t len, unsigned int seed)
+randomize_buffer(char *buf, size_t len, unsigned int seed)
 {
-    unsigned int *tmp = (unsigned int *)buf;
-    uint last;
-    long int remain = len;
-    int i;
     struct xrand xr;
+    uint64_t rand_val;
 
     if (len == 0)
         return;
 
     xrand_init(&xr, seed);
-    for (i = 0; remain > 0; i++, remain -= sizeof(*tmp)) {
-        if (remain > sizeof(*tmp)) { /* likely */
-            tmp[i] = xrand64(&xr);
-        } else { /* unlikely */
-            last = xrand64(&xr);
-            memcpy(&tmp[i], &last, remain);
-        }
+    for (size_t remain = len; remain > 0; remain -= MIN(remain, sizeof(rand_val))) {
+        rand_val = xrand64(&xr);
+        memcpy(buf + len - remain, &rand_val, MIN(remain, sizeof(rand_val)));
```

Comment:
You could ensure that memcpy() wont be generated for x86 by passing in a constant size:

```
for (...; remain >= sizeof(rand_val); remain -= sizeof(rand_val)) {
    memcpy(..., &rand_val, sizeof(rand_val));
```

then clean up the slop after the loop..

Reply:
Great point. Thanks Greg